<?xml version="1.0"?>

<launch>
	<include file="$(find openni_launch)/launch/openni.launch" />
	
	<!--Openni manager-->
	<!--node pkg = "nodelet" type = "nodelet" name = "openni_manager" output = "screen" respawn = "true" args = "manager" /-->
	<!--throttling-->
	<!--node pkg = "nodelet" type = "nodelet" name = "pointcloud_throttle" args = "load pointcloud_to_laserscan/CloudThrottle openni_manager">
		<param name = "max_rate" value = "2" />
		<remap from = "cloud_in" to="/camera/depth/points"/>
		<remap from = "cloud_out" to = "cloud_throttled" />
	</node-->
	<!--fake laser-->
	<!--node pkg="nodelet" type = "nodelet" name = "kinect_laser" args = "load pointcloud_to_laserscan/CloudToScan openni_manager">
		<param name = "output_frame_id" value = "/laser_link" />
		<remap from = "cloud" to = "cloud_throttled" />
	</node-->
	
	<remap from="image" to="/camera/depth/image_rect_raw"/>				<!--topic name of depth_image from openni-->
	<remap from="camera_info" to="/camera/depth/camera_info"/> 			<!--thong tin cau tao cua cam bien-->
	<node respawn = "true" name= "depthimage_to_laserscan" pkg="depthimage_to_laserscan" type="depthimage_to_laserscan">
		
		<param name = "scan_height" value = "60" type = "int" />		<!--chi so cua row ma ta muon lay lat cat-->
																						<!--scan_height cang lon thi lat cat can gan mat dat-->
		<param name = "output_frame_id" value = "/laser_link" type = "string" /> 	<!--day la he toa do cua lat cat-->
																											<!--mat dinh lat cat nam trong mat phang 0xy-->
		<param name = "scan_time" value = "0.04" type = "double" />	<!--tan so publish cua message tu cam bien laser-->
	</node>
</launch>
